src/main/java/ch/jalu/typeresolver/FieldUtils.java

```diff
@@ -0,0 +1,142 @@
+package ch.jalu.typeresolver;
+
+import ch.jalu.typeresolver.classutil.ClassUtils;
+
+import java.lang.reflect.Field;
+import java.lang.reflect.Modifier;
+import java.util.Arrays;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Optional;
+import java.util.function.Consumer;
+import java.util.function.Predicate;
+import java.util.stream.Collectors;
+
+/**
+ * Classes with utils for handling fields.
+ */
+public final class FieldUtils {
```

Comment:
I think the ConfigMe use case will be that I'll want fields, parents first, unique by name (with lowest in the hierarchy winning). Too specific or possible to do something?